### PR TITLE
:ambulance: web_website fixes after odoo update

### PR DESCRIPTION
**Pull request by KolushovAlexandr**

### CODE REVIEW

File: web_website/models/ir_property.py [hunk 1]
```diff
@@ -98,10 +100,86 @@ def get(self, name, model, res_id=False):
 
     @api.model
     def get_multi(self, name, model, ids):
-        return super(IrProperty, self._check_website_dependent(
-            name, model,
-            **GET_CONTEXT
-        )).get_multi(name, model, ids)
+        # Due to https://github.com/odoo/odoo/commit/b6d32de31e0e18a506ae06dc27561d1d078f3ab1 commit
+        # We override the super method to make it website dependent
```
Comment (yelizariev) [inline web_website/models/ir_property.py:107]: Mark code that is new and\or make overview of difference with origin

File: web_website/tests/test_website_dependent.py [hunk 2]
```diff
@@ -183,6 +183,8 @@ def test_website_dependent_priority(self):
             'value': 'only_website',
         })
         record.invalidate_cache()
+        # import wdb
+        # wdb.set_trace()
```
Comment (yelizariev) [inline web_website/tests/test_website_dependent.py]: :-1:

**Review by yelizariev — CHANGES_REQUESTED** [review]

File: web_website/models/ir_property.py [hunk 3]
```diff
@@ -98,10 +100,86 @@ def get(self, name, model, res_id=False):
...
+            result.update(res)
+
+        # set the default value to the ids that are not in result
+        default_value = fetched_result.pop()
+        for id in ids:
+            if id in result and (not result[id][3]):
+                # website_id  = result[3]
+                if default_value[3] or (
+                        result[id][3] != website_id and default_value[3] == website_id):
+                    result[id] = default_value
```
Comment (yelizariev) [inline web_website/models/ir_property.py:193]: It's hard to read. Use variabls, e.g.

website_id  = result[3]

**Review by yelizariev — CHANGES_REQUESTED** [review]

File: web_website/models/ir_property.py [hunk 4]
```diff
@@ -98,10 +103,111 @@ def get(self, name, model, res_id=False):
...
+                WHERE p.fields_id=%s
+                    AND (p.company_id=%s OR p.company_id IS NULL)
+                    AND (p.website_id=%s OR p.website_id IS NULL)
+                    AND (p.res_id IN %s OR p.res_id IS NULL)
+                ORDER BY p.website_id NULLS FIRST, p.company_id NULLS FIRST
+            """.format(comodel._table)
+            params = [model_pos, value_pos, field_id, company_id, website_id]
+        elif field.type in TYPE2FIELD:
+            model_pos = len(model) + 2
+            # retrieve values: p.res_id is formatted as "<rec._name>,<rec.id>"
```
Comment (yelizariev) [inline web_website/models/ir_property.py]: fix comment

File: web_website/models/ir_property.py [hunk 5]
```diff
@@ -98,10 +103,111 @@ def get(self, name, model, res_id=False):
...
+        # Due to https://github.com/odoo/odoo/commit/b6d32de31e0e18a506ae06dc27561d1d078f3ab1 commit
+        # We override the super method to make it website dependent
+        # It has the same idea and structure, but sql request and set record value method are changed
+        if not ids:
+            return {}
+        website_id = self._context.get('website_id', None)
+        field = self.env[model]._fields[name]
+        field_id = self.env['ir.model.fields']._get(model, name).id
+        company_id = (self._context.get('force_company')
+                      or self.env['res.company']._company_default_get(model, field_id).id)
```
Comment (yelizariev) [inline web_website/models/ir_property.py]: no need for parenthesises

File: web_website/models/ir_property.py [hunk 6]
```diff
@@ -98,10 +103,111 @@ def get(self, name, model, res_id=False):
...
+
+            res = {
+                i[0]: i for i in fetched_result
+            }
+            result.update(res)
+
+        # result format: {res_id: (res_id, val, company_id, website_id), ...}
+        # from fetched result was taken only the last row for each res_id
+
+        default_value = result.pop(None, None)
```
Comment (yelizariev) [inline web_website/models/ir_property.py:179]:
```suggestion
        # Take value for None key, i.e. 
        # {{None: (None, val0, company_id0, website_id0), ...}, res_id1: (res_id1, val1, company_id1, website_id1), ...}
        # when there is no such key, use value None
        default_value = result.pop(None, None)
```

**Comment by KolushovAlexandr** [comment]
11v branch should be updated at first

**Pull request closed without merging**